platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java

```diff
@@ -0,0 +1,1030 @@
+import org.apache.solr.client.solrj.StreamingResponseCallback;
+import org.apache.solr.client.solrj.beans.DocumentObjectBinder;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.SolrPingResponse;
+import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.SolrParams;
+import org.apache.solr.common.util.NamedList;
+
+/**
+ * Provides a pure interface to the Solr client allowing us to add new functionality.
+ *
+ * <p><i>Note:</i> This is a precursor to having the whole Solr client abstracted into interfaces to
+ * decouple us from the Solr implementation.
+ */
+public interface SolrClient extends Closeable {
+  /**
+   * Retrieve the corresponding SolrJ's client.
+   *
+   * @return the corresponding SolrJ's client
+   */
+  public org.apache.solr.client.solrj.SolrClient getClient();
+
+  /**
+   * Gets the core associated with this Solr client
+   *
+   * @return the core associated with this Solr client
+   */
+  public String getCore();
+
+  /**
+   * Checks whether the Solr server this client is connected to and the corresponding core is
+   * available or not. This method is different from an active ping as it might be basing its answer
+   * on cached information. It might, however, initiate a background ping.
+   *
+   * <p><i>Note:</i> Returning <code>true</code> doesn't guarantee the next call to any methods will
+   * succeed as the server or the core might become unavailable at any time without notice.
+   *
+   * @return <code>false</code> if the Solr server this client is connected or the corresponding
+   *     core is not available; <code>true</code> otherwise
+   */
+  public boolean isAvailable();
+
+  /**
+   * Waits if necessary for the Solr server this client is connected to and the corresponding core
+   * to become available. This method is different from an active ping as it might be basing its
+   * answer on cached information. It might, however, initiate a background ping.
+   *
+   * <p><i>Note:</i> Nothing will happen and the method will return right away if the client was
+   * already closed.
+   *
+   * <p><i>Note:</i> Returning <code>true</code> doesn't guarantee the next call to any methods will
+   * succeed as the server or the core might become unavailable at any time without notice.
+   *
+   * @param timeout the maximum time to wait
+   * @param unit the time unit of the timeout argument
+   * @return <code>false</code> if the Solr server this client is connected or the corresponding
+   *     core is not available; <code>true</code> otherwise
+   * @throws InterruptedException if the current thread was interrupted while waiting
+   */
+  public boolean isAvailable(long timeout, TimeUnit unit) throws InterruptedException;
```

Comment:
I am ok with removing the background ping portion.
⭕️ 
As for its use, it was mainly for the persistent store which is willing to wait for at least 5 seconds to issue the request instead of failing right away. So think of it as a way to issue the request with a timeout but instead of overloading all methods, they call this one first. Although you are absolutely right in the sense that the next call is not guarantee. It does account for starting up issues.

Reply:
👍 for the background ping